src/sortStudents.ts

```diff
@@ -1,16 +1,96 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const resultStudents: Student[] = [...students];
+  let averageA;
+  let averageB;
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  resultStudents.sort((a: Student, b: Student): number => {
+    switch (sortBy) {
+      case SortType.AverageGrade:
+        averageA = a.grades.reduce((sum: number, x: number) => sum + x)
+          / a.grades.length;
+
+        averageB = b.grades.reduce((sum: number, x: number) => sum + x)
+          / b.grades.length;
+
+        if (order === 'asc') {
+          return averageA - averageB;
+        }
+
+        if (order === 'desc') {
+          return averageB - averageA;
+        }
+        break;
+
+      case SortType.Name:
+        if (order === 'asc') {
+          return a.name.localeCompare(b.name);
+        }
+
+        if (order === 'desc') {
+          return b.name.localeCompare(a.name);
+        }
+        break;
+
+      case SortType.Surname:
+        if (order === 'asc') {
+          return a.surname.localeCompare(b.surname);
+        }
+
+        if (order === 'desc') {
+          return b.surname.localeCompare(a.surname);
+        }
+        break;
```

Comment:
```suggestion
      case SortType.Name:
      case SortType.Surname:
        return order === 'asc' 
          ? a[sortBy].localeCompare(b[sortBy]) 
           : b[sortBy].localeCompare(a[sortBy);
        break;
```
unite cases with same logic

Reply:
The sortBy variable contains the values ​​of SortType.Name and SortType.Surname. How to get to the Name and Surname properties of objects through this variable?